device/core/src/device_client.ts

```diff
@@ -96,7 +96,7 @@ export class Client extends InternalClient {
       if (err && this._retryPolicy.shouldRetry(err)) {
         debugErrors('reconnect policy specifies a reconnect on error');
         /*Codes_SRS_NODE_DEVICE_CLIENT_16_097: [If the transport emits a `disconnect` event while the client is subscribed to c2d messages the retry policy shall be used to reconnect and re-enable the feature using the transport `enableC2D` method.]*/
-        if (this._c2dFeature) {
+        if (this._userRegisteredC2dListener) {
```

Comment:
>_userRegisteredC2dListener [](http://example.com/codeflow?start=17&length=26)

<no action required> It hurts my soul to see `this._enableXXX` code inside the `_deviceDisconnectHandler` function

Reply:
I'm going to leav it.  I'm assuming it hurts your soul because it should have resided in some renablefeature class?

Reply:
I have no problem leaving it as-is.

It hurts my soul because it looks like this code is trying to enable C2D as part of the disconnect process. This is odd. Plus, it's not obvious that `this._enableC2D` is reconnecting the transport.

In other words, enabling C2D is a side-effect of being disconnected, and reconnecting is a side-effect of Enabling C2D. 

If C2D is not enabled, then the reconnect happens in a completely different place.  I hope.